### ISSUE TITLE
Styles not working for Stylised used in ask

### ISSUE THREAD

**Issue by matchwood** [pjones/byline#1]
The example given on Hackage:

```
let question = "What's your favorite " <> ("language" <> bold)    <> "? "
ask question Nothing
```

does not produce stylised text (compiled with Stack nightly-2016-04-20). `say` is working fine, so I think it is something to do with the [renderText](https://github.com/pjones/byline/blob/master/src/System/Console/Byline/Internal/Render.hs#L72) function. If I have time I'll play around with this a little to see if I can work out what is going wrong.

**Comment by samtay** [comment]
I've tracked this down to the `defRenderMode` function, specifically querying for terminal info in the sub-function [`getMaxColors`](https://github.com/pjones/byline/blob/6cc4fbfe32c1d69692cb409ac12350474a63772c/src/System/Console/Byline/Internal/Byline.hs#L85-L92).

From what I can tell, this either isn't a bug and it's an actual terminal specific issue, or the bug resides in one of the dependencies such as `terminfo-hs`.

It's odd because even if I change the code to force a return value such that `(sayMode, askMode) == (Simple, Simple)` or `(sayMode, askMode) == (Term256, Term256)`, I still don't get the correct styling. This leads me to believe it is a specific terminal issue (I'm on xterm-termite).

Then again, I still have no idea why we need different modes for `ask` and `say`, since `ask` is, at a low level, just printing a prompt to stdout the same way `say` should. I'm very confused.

**Comment by samtay** [comment]
Also, forgot to mention a very important note (and workaround at least for me), which is that specifying specific rgb values does style successfully. To see if its the same for you, just try running the above except replace
```haskell
"language" <> bold
```
with
```haskell
"language" <> bold <> fg (rgb 108 113 196)
```
and see if you get purple text.

Now *that* makes it seem like a byline issue. Very odd all around.

**Comment by pjones** [comment]
The reason there are different render modes for `say` and `ask` is due to the fact they are handled differently.  When using `say`, byline can write the message directly to the handle (usually STDOUT).  But for `ask`, byline sends the prompt text to `Haskeline` after adding proper escape codes to the prompt.

I suspect that [getInputLine](https://hackage.haskell.org/package/haskeline-0.7.4.0/docs/System-Console-Haskeline.html#v:getInputLine) isn't dealing with the escape codes correctly.

I'm very busy right now so if someone wants to help track this down I would appreciate it.

**Comment by pjones** [comment]
It turns out that this was a really stupid bug in Byline.  I found and fixed it while cleaning up the rendering code.  The fix will be in the upcoming 1.0 release.
